src/preact/base-element.js

```diff
@@ -321,12 +339,23 @@ function collectProps(Ctor, element, defaultProps) {
   // as separate properties. Thus in a carousel the plain "children" are
   // slides, and the "arrowNext" children are passed via a "arrowNext"
   // property.
+  const errorMessage =
+    'only one of "passthrough", "passthroughNonEmpty"' +
+    ' or "children" may be given';
   if (passthrough) {
-    devAssert(
-      !childrenDefs,
-      'only one of "passthrough" or "children" may be given'
-    );
+    devAssert(!childrenDefs && !passthroughNonEmpty, errorMessage);
     props['children'] = [<Slot />];
+  } else if (passthroughNonEmpty) {
+    devAssert(!childrenDefs, errorMessage);
+    // If all children are whitespace text nodes, consider the element as
+    // having no children
+    props['children'] = element
+      .getRealChildNodes()
+      .every(
+        (node) => node.nodeType === 3 && node.nodeValue.trim().length === 0
```

Comment:
Add a comment here that `nodeType === 3` indicates `TEXT_NODE`. We usually do this for most of constants like this: `node.nodeType == /* TEXT_NODE */ 3`.

Reply:
Thank you!  I have added!